### PR TITLE
[ISSUE #3805] sub task3 add @ShenyuService for `shenyu-client-alibaba-dubbo`

### PR DESCRIPTION
**Pull request by mangoGoForward**
Signed-off-by: mango <xu.weiKyrie@foxmail.com>

#3805

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by dragon-zhang** [comment]
@mangoGoForward please add wechat: Z18179469818

File: shenyu-client/shenyu-client-dubbo/shenyu-client-dubbo-common/pom.xml [hunk 1]
```diff
@@ -31,5 +31,10 @@
...
         </dependency>
+        <dependency>
+            <groupId>com.alibaba</groupId>
+            <artifactId>dubbo</artifactId>
+            <version>${alibaba.dubbo.version}</version>
+        </dependency>
     </dependencies>
-</project>
\ No newline at end of file
+</project>
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-dubbo/shenyu-client-dubbo-common/pom.xml:40]: should not add `alibaba-dubbo dependency` to `shenyu-client/shenyu-client-dubbo/shenyu-client-dubbo-common/pom.xml`.

File: shenyu-client/shenyu-client-dubbo/shenyu-client-dubbo-common/src/main/java/org/apache/shenyu/client/dubbo/common/annotation/ShenyuService.java [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+
+
+/**
+ * The interface shenyu service.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD})
+@Service
+@Inherited
+public @interface ShenyuService {
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-dubbo/shenyu-client-dubbo-common/src/main/java/org/apache/shenyu/client/dubbo/common/annotation/ShenyuService.java:39]: This class should move to `shenyu-client-alibaba-dubbo` module, and it's seems that you lost attritubes in`@ShenyuDubboClient`.

**Comment by dragon-zhang** [comment]
Only the examples in `shenyu-examples-alibaba-dubbo-service-annotation` should be changed, as for other example modules, please rollback them.

File: shenyu-client/shenyu-client-dubbo/shenyu-client-dubbo-common/pom.xml (hunk 1 above)
Comment (mangoGoForward) [reply shenyu-client/shenyu-client-dubbo/shenyu-client-dubbo-common/pom.xml]: Done

File: shenyu-client/shenyu-client-dubbo/shenyu-client-dubbo-common/src/main/java/org/apache/shenyu/client/dubbo/common/annotation/ShenyuService.java (hunk 2 above)
Comment (mangoGoForward) [reply shenyu-client/shenyu-client-dubbo/shenyu-client-dubbo-common/src/main/java/org/apache/shenyu/client/dubbo/common/annotation/ShenyuService.java]: Done

File: shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/annotation/ShenyuService.java [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+@Inherited
+@ShenyuDubboClient
+public @interface ShenyuService {
+
+    /**
+     * Path string.
+     *
+     * @return the string
+     */
+    @AliasFor(annotation = Component.class)
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/annotation/ShenyuService.java:48]: change to `@AliasFor(annotation = ShenyuDubboClient.class)`.

File: shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/annotation/ShenyuService.java [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+    @AliasFor(annotation = Service.class)
+    String group() default "";
+
+    /**
+     * Alias for {@link Service#path()}.
+     *
+     * @return the string
+     */
+    @AliasFor(annotation = Service.class)
+    String path() default "";
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/annotation/ShenyuService.java:113]: change to `@AliasFor(annotation = ShenyuDubboClient.class)` and add a `dubboPath` method.

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/alibaba/dubbo/service/annotation/impl/DubboClassMultiParamServiceImpl.java [hunk 5]
```diff
@@ -33,7 +33,7 @@
  * The type Dubbo multi param service.
  */
 @ShenyuDubboClient("/demo")
-@Service
+@ShenyuService(value = "ShenyuExampleApp.ShenyuExampleServer.AlibabaDubboClass.MultiParam")
 public class DubboClassMultiParamServiceImpl implements DubboClassMultiParamService {
```
Comment (dragon-zhang) [inline shenyu-examples/shenyu-examples-dubbo/shenyu-examples-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/alibaba/dubbo/service/annotation/impl/DubboClassMultiParamServiceImpl.java:37]:
```java
@ShenyuDubboClient("/demo")
@ShenyuService(value = "ShenyuExampleApp.ShenyuExampleServer.AlibabaDubboClass.MultiParam")
```
should change to :
```java
@ShenyuService("/demo")
```
the other impls also need to change.

File: shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/annotation/ShenyuService.java (hunk 3 above)
Comment (mangoGoForward) [reply shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/annotation/ShenyuService.java]: Done. thx

File: shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/annotation/ShenyuService.java (hunk 4 above)
Comment (mangoGoForward) [reply shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/annotation/ShenyuService.java]: Done.

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/alibaba/dubbo/service/annotation/impl/DubboClassMultiParamServiceImpl.java (hunk 5 above)
Comment (mangoGoForward) [reply shenyu-examples/shenyu-examples-dubbo/shenyu-examples-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/alibaba/dubbo/service/annotation/impl/DubboClassMultiParamServiceImpl.java:36]: Done

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/alibaba/dubbo/service/annotation/impl/DubboTestServiceImpl.java [hunk 6]
```diff
@@ -29,7 +29,7 @@
 /**
  * The type Dubbo service.
  */
-@Service
+@ShenyuService("demo")
```
Comment (dragon-zhang) [inline shenyu-examples/shenyu-examples-dubbo/shenyu-examples-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/alibaba/dubbo/service/annotation/impl/DubboTestServiceImpl.java:32]: rollback this.

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/alibaba/dubbo/service/annotation/impl/DubboMultiParamServiceImpl.java [hunk 7]
```diff
@@ -31,7 +31,7 @@
 /**
  * The type Dubbo multi param service.
  */
-@Service
+@ShenyuService("demo")
 public class DubboMultiParamServiceImpl implements DubboMultiParamService {
```
Comment (dragon-zhang) [inline shenyu-examples/shenyu-examples-dubbo/shenyu-examples-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/alibaba/dubbo/service/annotation/impl/DubboMultiParamServiceImpl.java:35]: rollback this.

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/alibaba/dubbo/service/annotation/impl/DubboTestServiceImpl.java (hunk 6 above)
Comment (mangoGoForward) [reply shenyu-examples/shenyu-examples-dubbo/shenyu-examples-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/alibaba/dubbo/service/annotation/impl/DubboTestServiceImpl.java]: Done.

**Comment by dragon-zhang** [comment]
@mangoGoForward have fun with contribution !

**Review by dragon-zhang — APPROVED** [review]

**Pull request merged**